### PR TITLE
Add Prow like actions for BYOH

### PR DESCRIPTION
**Pull request by shamsher31**
**What this PR does / why we need it**:
This will enable Prow like actions to interact with PR and issues.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/246

**Additional information**
Prow Github Action: https://github.com/jpmcb/prow-github-actions
Labeler: https://github.com/actions/labeler

**Special notes for your reviewer**
You can use commands like:
/assign @anusha94 
/cc @jamiemonserrate 
/kind enhancement
/area installer
/approve
/lgtm
/hold
/close 
/reopen 

See https://github.com/kubernetes/kubernetes/pulls for more info.

### CODE REVIEW

File: .github/workflows/lgtm.yaml
Comment (anusha94) [inline .github/workflows/lgtm.yaml:1]: I'm assuming the label will take effect when folks with the right permissions add it, right?

File: .github/workflows/prow-github-actions.yaml
Comment (anusha94) [inline .github/workflows/prow-github-actions.yaml:13]: nit: I'd prefer them to alphabetically sorted if possible :)

**Review by anusha94 — COMMENTED** [review]
Thanks a bunch for this PR!

File: .github/workflows/lgtm.yaml
Comment (shamsher31) [reply .github/workflows/lgtm.yaml:1]: Right, only those who have sufficient permissions will take effect, also we have branch protection rules enabled so PR's won't get merged unless all the checks are passed.

File: .github/labeler.yaml
Comment (dharmjit) [inline .github/labeler.yaml:7]: Suggestion: The label `area/config` sounds  little confusing to me because here we are only keeping CRDs, RBAC manifests. I guess maybe this is not required at all or we could label it `area/apis`. @anusha94 @jamiemonserrate

File: .github/labeler.yaml
Comment (shamsher31) [reply .github/labeler.yaml:10]: Is this renamed in some recent PR? I don't see the folder `agent/installer`. But I will update it.

File: .github/workflows/prow-github-actions.yaml
Comment (shamsher31) [reply .github/workflows/prow-github-actions.yaml:13]: Yup, updated.

File: .github/labeler.yaml
Comment (dharmjit) [inline .github/labeler.yaml:20]: Suggestion: Do we require this label as well?

File: .github/labeler.yaml
Comment (dharmjit) [inline .github/labeler.yaml:36]: Suggestion: I guess we do not require 1-1 mapping of the label area to label kind. It could be another list with values something like below.
- bug
- feature
- cleanup

File: .github/workflows/lgtm.yaml
Comment (dharmjit) [inline .github/workflows/lgtm.yaml:8]: @jamiemonserrate @anusha94 Do we require Auto Merge?

**Review by dharmjit — COMMENTED** [review]
Awesome Work @shamsher31. LGTM, only some suggestions/questions.

File: .github/labeler.yaml
Comment (shamsher31) [reply .github/labeler.yaml:20]: Not super important for now but this will help us identify and filter PR's which are related to either releases or testing. We can later assign people who are experts in that area. Any changes to `go.mod` will directly affect the release and it's important to label them so we can filter and take actions accordingly.

File: .github/labeler.yaml
Comment (shamsher31) [reply .github/labeler.yaml:36]: Make sense. What else we can add?

File: .github/labeler.yaml
Comment (anusha94) [reply .github/labeler.yaml:7]: Hmmm makes sense. I'm okay to not having config label.

File: .github/labeler.yaml
Comment (anusha94) [reply .github/labeler.yaml:20]: +1 to keeping this label.

File: .github/workflows/lgtm.yaml
Comment (anusha94) [reply .github/workflows/lgtm.yaml:8]: Again, I'm nervous about it. No reason. Just nervous 😨

File: .github/labeler.yaml
Comment (shamsher31) [reply .github/labeler.yaml:7]: I'm fine removing the `area/config` label as it won't be updated frequently

As per my understanding `area/apis` should only consist of pure contracts and nothing else. We already have a dedicated label for `apis`.

File: .github/workflows/lgtm.yaml
Comment (shamsher31) [reply .github/workflows/lgtm.yaml:8]: Ok, let me clarify a few points that will clear your doubts.

1. The `/lgtm` label does not do anything on its own, it just signals that you've finished reviewing a PR and are satisfied with the changes. No merge will take place when you say `/lgtm` unless you approve the PR.
2. Once the reviewer is satisfied with changes, they can say `/approve`. If they want one more eye to go through the changes they can request another reviewer.
3. The best part is, we have branch protections enabled for PR merge. What does it mean, unless all the review comments are resolved, and the PR got 2 approvals, none of these labels will take precedence.
4. As the owner and reviewer of the code repository, we never blindly approve PR, we take precautions and go through the changes carfully, we only approve when we think everything will work fine (some exceptions, because we are human :))
5. The `auto-merge` will only be enabled if you click on the `Enable auto-merge(squash)` button as shown in this PR. This means that once all the criteria(test passes, all comments resolved, 2 approvals) to merge the PR is fulfiled, the PR will merge after that and you don't have to come again to merge the PR manually.

Please let me know if you want more details.

Comment (anusha94) [reply .github/workflows/lgtm.yaml:8]: Convinced! +1 🙂

**Review by dharmjit — APPROVED** [review]

**Review by anusha94 — APPROVED** [review]
lgtm!

**Comment by jpmcb** [comment]
Glad to see this being used out in the wild! Feel free to ping me with any questions using the prow bot! ❤️

**Pull request merged**
